docker/kopia-build/Dockerfile

```diff
@@ -0,0 +1,45 @@
+FROM golang:1.17-alpine3.14 AS builder
+
+ARG kopiaBuildCommit
+ARG kopiaRepoOrg
+
+RUN apk add git
+
+WORKDIR /
+
+RUN git clone https://github.com/${kopiaRepoOrg}/kopia.git
+
+ENV GITHUB_REPOSITORY=https://github.com/${kopiaRepoOrg}/kopia
+
+WORKDIR /kopia
+
+# Build kopia binary from specific commit
+RUN git checkout ${kopiaBuildCommit}
+# Refers to goreleaser/goreleaser:v1.11.2
+COPY --from=docker.io/goreleaser/goreleaser@sha256:b13418f20019fffb29797aff3d03f5e9ca84cea93634f4169a7ed603a95ab198 \
```

Comment:
Is this the same goreleaser version used during release?

Reply:
In kanister build, we are using `goreleaser/goreleaser:v1.2.4`. It's about 8 months old when we last updated it.

Reply:
@ankitjain235 (@pavannd1) Can this be simplified?

For example, does `goreleaser` need to be used here?

It is possible to build kopia directly. For example, see https://github.com/kastenhq/kopia/blob/publish/docker/kopia-alpine/Dockerfile

A few questions:
- is is necessary to build a container image for kopia? or would it suffice to build the binary, say as part of building the tools image?
- is it necessary to build the image or binary every time? or can it be cached and reused, and only rebuilt when the kopia version / commit id changes?

Reply:
@julio-lopez 

We used goreleaser just so that we can build the kopia binary the way it is done upstream and we won't have to worry about changing the command if it needs to be changed any time as per change upstream.

In a follow-up PR(https://github.com/kanisterio/kanister/pull/1620) that is WIP, I am adding an action that can be manually triggered whenever we want to build kopia. I kept the process to build a container image for kopia only for this use case. That way, we would only build Kopia image when it's required. Making it as a part of kanister-tools image would result in re-building the binary in scenarios I tried out.